### ISSUE TITLE
What is the difference between the trainings with a pretrained model? When should you use which?

### ISSUE THREAD

**Issue by dmddmd** [ultralytics/ultralytics#1786]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

There are 2 ways in the docs: 
```
# Start training from a pretrained *.pt model
yolo detect train data=coco128.yaml model=yolov8n.pt epochs=100 imgsz=640

# Build a new model from YAML, transfer pretrained weights to it and start training
yolo detect train data=coco128.yaml model=yolov8n.yaml pretrained=yolov8n.pt epochs=100 imgsz=640
```

Why?

**Comment by Adonis2018** [comment]
What is the difference between these two commands? They both seem to use pre-trained models. @AyushExel

**Comment by glenn-jocher** [comment]
@Adonis2018 both commands use pre-trained models and train them further on a specific dataset. However, the difference between the two is the way the pre-trained model is used. 

The first command simply uses a pre-trained model as is and starts training on a new dataset. The pre-trained model has already learned to identify certain patterns and features in images, so it brings that knowledge into the new training process. 

The second command builds a new model based on a "config" file (a YAML file) and then initializes it with weights from the pre-trained model. This allows you to customize the structure of the model to better suit your specific dataset, while still leveraging the pre-trained weights. 

In summary, the first command uses a pre-trained model as is, while the second command builds a new model based on a config file and initializes it with weights from a pre-trained model.

**Comment by Adonis2018** [comment]
> 
Thank you！
@glenn-jocher

**Comment by glenn-jocher** [comment]
@Adonis2018 you're welcome! I'm glad I could assist you. When it comes to building machine learning models, there are different ways to use pre-trained models. One way is to use the pre-trained model as is and start training on a new dataset. Another way is to build a new model based on a config file and then initialize it with weights from a pre-trained model. The latter approach allows you to customize the structure of the new model to better suit your specific dataset, while still leveraging the knowledge learned by the pre-trained weights.

**Comment by ExtReMLapin** [comment]
Sharing my grain of salt here.

I used yolov5 to detect Minutiaes, fingerprints and diatoms.

Diatoms segmentation with worked much better when doing transfer learning as a diatom has an actual "frontier".
![image](https://user-images.githubusercontent.com/3909752/230013293-5f4f6b59-fc83-416e-8cd6-d8530d14eb48.png)


On another end, with fingerprint zones, and minutiaes it's more complex, freezing the backbone gives bad results, so I had to train from scratch as there is no such "clear" frontier on what is a minutiae.


![image](https://user-images.githubusercontent.com/3909752/230026931-6c30918d-6299-4622-aa16-fcb15176c10c.png)

**Comment by glenn-jocher** [comment]
@Adonis2018 that's a great point! When it comes to using transfer learning with pre-trained models, it's important to consider the nature of the objects you are detecting. If the objects have a clear boundary or structure, like the diatoms you mentioned, transfer learning can work well. However, if the objects don't have a clear boundary or structure, like the minutiae and fingerprints you mentioned, training from scratch may be necessary to get good results. 

This is because the backbone of a pre-trained model is often designed to detect certain patterns and features in the data, and if the objects you're detecting don't fit those patterns or lack clear boundaries, the pre-trained weights may not be useful. In these cases, building a new model from scratch allows you to design the architecture to better suit the specific characteristics of the objects you're detecting.

**Comment by rgaufman** [comment]
This thread is really useful and all the replies are much appreciated. I do have some remaining unknowns though:

1. How much slower is building a model from scratch? - it's taken my M1 around 4 hours to train with my training set of images starting with the [yolov8s.pt](http://yolov8s.pt/) model. Are we talking double the time? 10x the time?
2. My application is detecting small objects of trash in a large 4K image, example below. My training set is the excellent [tacodataset.org](http://tacodataset.org/) - I am getting mixed results though with sometimes people's heads being mistaken as trash etc. Is this an example of needing to train from scratch? - or maybe using a totally different approach?

![15052023-0919AM-0920AM@0min2s](https://github.com/ultralytics/ultralytics/assets/33842342/6bc92a4b-ae2a-455f-9522-526af18b5173)

**Comment by ExtReMLapin** [comment]
While it's not possible yet to [freeze the backbone](https://github.com/ultralytics/ultralytics/issues/793) I would suggest you for your kind of work to freeze the backbone (when it's possible), there is two reasons behind it 

* Faster training
* Better backbone knowledge as it's training on COCO dataset, which has much more image that you will have to train the model on (from scratch).

**Comment by glenn-jocher** [comment]
@rgaufman thanks for reaching out to us! While it's not currently possible to freeze the backbone of the model in YOLOv8, when it becomes available, freezing the backbone can offer some benefits for your use case. 

Freezing the backbone would allow you to use the pre-trained weights from the COCO dataset, which would provide a better initial knowledge of the features and patterns to look for in images. This can lead to faster training times, as well as better results if the pre-trained weights are a good fit for your specific dataset.

However, in cases where objects in your images don't have clear boundaries or structures, training from scratch may be necessary to achieve good results. From the example image you have given, it seems that this could be the case, as the trash objects may not have clearly defined boundaries. 

In addition to training from scratch, there are also other approaches you can take to improve your results, such as adjusting your model architecture, tuning hyperparameters, or using a different pre-trained model if available. 

Again, thanks for using YOLOv8, and please let us know if you have any other questions or concerns!

**Comment by sech-ai** [comment]
Tengo que detectar arboles de imágenes obtenidas por drones , estoy utilizando yolov8, el cuál realice la una base de datos de 500 imágenes, procese la imágenes. pero llega el momento de realizar la detección y no me detecta ningún árbol
Para realizar el entrenamiento ocupe esto comando: yolo detect train model=yolov8n.pt data=data\arbol_final.yaml imgsz=640 workers=0 batch=2 device=0 epochs=100 patience=50 name=yolov8_proces_final_arbol
Para realizar la deteccion esto comando:yolo detect predict model=runs\detect\yolov8_proces_final_arbol8\weights\best.pt  source=inference\arbol_2.JPG save=True
![resultado](https://github.com/ultralytics/ultralytics/assets/132520086/dee5bcff-0293-4158-bdf6-e76b18d79de0)

**Comment by glenn-jocher** [comment]
@sech-ai gracias por utilizar YOLOv8 para tu proyecto de detección de árboles en imágenes de drones. Parece que estás enfrentando un problema donde el modelo no detecta ningún árbol después de entrenar con una base de datos de 500 imágenes.

Hay algunas posibles razones por las que esto podría estar sucediendo. Primero, asegúrate de que tus imágenes de entrenamiento contengan una variedad adecuada de ejemplos de árboles y estén etiquetadas correctamente. Si la base de datos de entrenamiento no incluye suficientes ejemplos de árboles o si las etiquetas están incorrectas, el modelo puede tener dificultades para aprender a detectarlos.

Además, verifica si el tamaño de imagen que estás utilizando (640x640) es adecuado para detectar árboles en tus imágenes. Dependiendo del tamaño y la resolución de los árboles en las imágenes de drones, es posible que necesites ajustar el tamaño de imagen para mejorar la detección.

También es importante verificar la configuración de entrenamiento utilizada, como el número de lotes (batch size), la cantidad de épocas (epochs), la paciencia (patience) y otros parámetros. Ajustar estos parámetros adecuadamente puede tener un impacto significativo en el rendimiento del modelo.

Te recomendaría revisar los ejemplos de detección en el conjunto de entrenamiento para verificar si las etiquetas son correctas y si los árboles son reconocibles por el modelo. Además, puedes experimentar con diferentes tamaños de imagen y configuraciones de entrenamiento para ver si mejora la detección de árboles en tus imágenes de drones.

Es importante recordar que el rendimiento del modelo puede mejorar con un conjunto de datos más grande y diverso, así como con ajustes finos específicos para tu tarea de detección de árboles.

¡Espero que estos consejos te sean útiles y te deseo mucho éxito en tu proyecto de detección de árboles!

**Comment by LichiV** [comment]
Dear @glenn-jocher thank you very much for your extensive support and the many answers from the Github forum. I am investigating transfer learning for my final thesis. Interestingly, there are classes that perform better than other classes after freezing the backbone and other layers. The other classes are small objects that also contain little pixel information. If I do not freeze any layers, the small classes are still recognized well. 
You have already described above that it could be due to unclear contours of the small objects. Of course I would also like to include this in my work, but I can't find any similar studies in the literature where TL works well with certain classes and not with others. Do you happen to have any idea where I could get such citable information?
That would be very interesting. 
Thank you

**Comment by Petros626** [comment]
@AyushExel @glenn-jocher whats the difference between these two and, which performs generally better?

```python
model = YOLO("yolo11n.pt")  # load a pretrained model (recommended for training)
model = YOLO("yolo11n.yaml").load("yolo11n.pt")  # build from YAML and transfer weights
```

**Comment by ExtReMLapin** [comment]
It's all the same.

It's only useful to load yaml first if the .pt doesn't exactly match (example : head)

Example, transfer learning from detection model to segmentation yaml

**Comment by Petros626** [comment]
Thanks for the quick answer. 

I guess when I try to transfer-learn from camera images (e.g. COCO) to bird's eye view images (from 3d-lidar) the transfer weights approach suits better?

**Comment by glenn-jocher** [comment]
Yes, transferring weights via YAML (`model=yolo11n.yaml + pretrained=yolo11n.pt`) is generally better when adapting to significantly different input domains like bird's eye view from LiDAR. This approach allows architectural modifications while preserving transferred feature knowledge from COCO. For best results, experiment with both methods and monitor validation metrics, as effectiveness can vary based on dataset alignment. For more details, see the [Transfer Learning](https://www.ultralytics.com/glossary/transfer-learning) and [Fine-tuning](https://www.ultralytics.com/glossary/fine-tuning) guides.
